### PR TITLE
feat(access): add AccessControl e2e tests

### LINKED ISSUE #175: Access Control e2e tests

### PR DESCRIPTION
**Pull request by alexfertel**
Resolves #175 

#### PR Checklist

- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: examples/access-control/tests/access_control.rs [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+
+    let AccessControl::getRoleAdminReturn { role } =
+        contract.getRoleAdmin(DEFAULT_ADMIN_ROLE.into()).call().await?;
+    assert_eq!(*role, DEFAULT_ADMIN_ROLE);
+
+    Ok(())
+}
+
+#[e2e::test]
+async fn non_admin_cannot_grant_role(
```
Comment (bidzyyys) [inline examples/access-control/tests/access_control.rs:73]: We agreed on `error_when_<function>_<description>` test name, check #155

File: examples/access-control/tests/access_control.rs [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+        role: ROLE.into(),
+        account: bob_addr,
+        sender: alice_addr
+    }));
+
+    Ok(())
+}
+
+#[e2e::test]
+async fn non_admin_cannot_revoke_role(
```
Comment (bidzyyys) [inline examples/access-control/tests/access_control.rs:160]: Same comment as above regarding test name.

File: examples/access-control/tests/access_control.rs [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+        role: ROLE.into(),
+        account: bob_addr,
+        sender: bob_addr
+    }));
+
+    Ok(())
+}
+
+#[e2e::test]
+async fn only_sender_can_renounce(alice: Account, bob: Account) -> Result<()> {
```
Comment (bidzyyys) [inline examples/access-control/tests/access_control.rs:241]: And here.

File: examples/access-control/tests/access_control.rs [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+        role: ROLE.into(),
+        account: alice_addr,
+        sender: bob_addr
+    }));
+
+    Ok(())
+}
+
+#[e2e::test]
+async fn previous_admins_no_longer_grant_roles(
```
Comment (bidzyyys) [inline examples/access-control/tests/access_control.rs:361]: And here.

File: examples/access-control/tests/access_control.rs [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+    assert!(err.reverted_with(AccessControlUnauthorizedAccount {
+        account: alice_addr,
+        neededRole: NEW_ADMIN_ROLE.into()
+    }));
+
+    Ok(())
+}
+
+#[e2e::test]
+async fn previous_admins_no_longer_revoke_roles(
```
Comment (bidzyyys) [inline examples/access-control/tests/access_control.rs:390]: And here.

**Review by bidzyyys — APPROVED** [review]
I like it 🚀  
My comments are connected with #155

**Review by bidzyyys — APPROVED** [review]
LGTM 🚀

**Review by qalisander — APPROVED** [review]
Good job!

**Pull request merged**